postprocessing/science/compute_isochrones.py

```diff
@@ -0,0 +1,250 @@
+
+parser.add_argument('--events', nargs=1, choices=[0,1,2], default=([0]), 
+                    help='0: xdmf file contains just one event; 1: process the first of two events; 2: process the second of two events' ,type=int) 
+# Needed for Ridgecrest output, where the xdmf file contains two events 
+
+args = parser.parse_args()
+
+
+def ComputeTriangleMidpoints(geom, connect):
+    """Generates an array with coordinates of triangle midpoints (same dimension as connect)"""
+    xyz = np.zeros_like(connect, dtype=float)
+    xyz = (1./3.)*(geom[connect[:,0],:]+geom[connect[:,1],:]+geom[connect[:,2],:])   
+    return xyz
+
+
+def GetTimeIndices(xdmfFile):
+    """Selects the correct time steps for output that contains two events"""    
+    if args.events[0] == 0:
+        return [0, xdmfFile.ReadNdt()]
+    elif args.events[0] == 1:
+        return [0, int(xdmfFile.ReadNdt()/2)]
+    else:
+        return [int(xdmfFile.ReadNdt()/2), xdmfFile.ReadNdt()]
+
+
+def LoadSliprates(sliprateParameters):
+    sliprates = []
+    if "SRs" in sliprateParameters:
+        sliprates.append(faultXdmf.ReadData("SRs")[timeIndicesFault[0]:timeIndicesFault[1]].T)
+    if "SRd" in sliprateParameters:
+        sliprates.append(faultXdmf.ReadData("SRd")[timeIndicesFault[0]:timeIndicesFault[1]].T)
+    if "ASR" in sliprateParameters:
+        if "SRs" in sliprateParameters and "SRd" in sliprateParameters:
+            sliprates.append(np.sqrt(sliprates[0]**2 + sliprates[1]**2))
+        else:
+            sliprates.append(np.sqrt(faultXdmf.ReadData("SRs")[timeIndicesFault[0]:timeIndicesFault[1]].T**2 +
+                                     faultXdmf.ReadData("SRd")[timeIndicesFault[0]:timeIndicesFault[1]].T**2))
+    return np.array(sliprates)
+
+
+def ClosestReceiverIndex(coords, receiver_array):
+    dist = np.sum((receiver_array-coords)**2, axis=1)
+    return np.argmin(dist)
+
+
+def PTraveltime(receiverIndex, trigger=0.001):
+    """Picks the P-wave arrival. This function is only suited for synthetic data without noise"""
+    
+    timeIndices = GetTimeIndices(surfaceXdmf)
+    absWaveform = np.sqrt(surfaceXdmf.ReadDataChunk(surfaceVariables[0], receiverIndex, 1)**2 +
+                      surfaceXdmf.ReadDataChunk(surfaceVariables[1], receiverIndex, 1)**2 +
+                      surfaceXdmf.ReadDataChunk(surfaceVariables[2], receiverIndex, 1)**2)[timeIndices[0]:timeIndices[1]]
+    
+    threshold = np.amax(absWaveform)*trigger           # trigger dictates the portion of the maximum ground velocity,
+    t = np.argmax(absWaveform>threshold)*dtSurface     # which needs to be reached to pick the p-phase
+    return t                                           # p-phase arrival time (equal to traveltime if nucleation at t=0)
+
+
+def ApproximateHypocenter(absoluteSliprates, faultCells, slipRateThreshold=args.sliprateThreshold[0]):
+    
+    slippingFault = np.where(absoluteSliprates > slipRateThreshold, 1, 0)
```

Comment:
wouldn't it be more simple to use the min of RT variable?

Reply:
yes, I rewrote the function